incubator/sparkoperator/templates/webhook-cleanup-job.yaml

```diff
@@ -5,7 +5,7 @@ metadata:
   name: {{ include "sparkoperator.fullname" . }}-webhook-cleanup
   namespace: {{ .Release.Namespace }}
   annotations:
-    "helm.sh/hook": pre-delete, pre-upgrade
```

Comment:
Can you explain why the secret shouldn't be cleaned up before an upgrade?

Reply:
The init job handles the secret already existing. While, I don't think it's dangerous to delete the secret, it does cause some unnecessary churn. While deleting the secret could be (accidentally) helpful by extending the expiration of the certificate, I feel like it would be better to do that explicitly somewhere else.

Reply:
I think we should leave the original hooks here. The init job won't create a new secret if one already exists, which is not ideal.